private/tools/java/rules/jvm/external/maven/Outdated.java

```diff
@@ -158,28 +256,61 @@ public static void main(String[] args) throws IOException {
       String artifactId = artifactParts[1];
       String version = artifactParts[2];
 
-      String releaseVersion = null;
+      ArtifactReleaseInfo artifactReleaseInfo = null;
       for (String repository : repositories) {
-        releaseVersion = getReleaseVersion(repository, groupId, artifactId);
-        if (releaseVersion != null) {
+        artifactReleaseInfo = getReleaseVersion(repository, groupId, artifactId);
+
+        if (artifactReleaseInfo != null) {
+          // We return the result from the first repository instead of searching all repositories
+          // for the artifact
           verboseLog(
               String.format(
-                  "Found version [%s] for %s:%s in %s",
-                  releaseVersion, groupId, artifactId, repository));
-          // Should we search all repositories in the list for latest version instead of just the
-          // first
-          // repository that has a version?
+                  "Found release version [%s] and pre-release version [%s] for %s:%s in %s",
+                  artifactReleaseInfo.releaseVersion,
+                  artifactReleaseInfo.preReleaseVersion,
+                  groupId,
+                  artifactId,
+                  repository));
           break;
         }
       }
 
-      if (releaseVersion == null) {
+      if (artifactReleaseInfo == null) {
         verboseLog(String.format("Could not find version for %s:%s", groupId, artifactId));
-      } else if (new ComparableVersion(releaseVersion).compareTo(new ComparableVersion(version))
-          > 0) {
-        System.out.println(
-            String.format("%s:%s [%s -> %s]", groupId, artifactId, version, releaseVersion));
-        foundUpdates = true;
+      } else {
+        if (artifactReleaseInfo.hasPreReleaseVersionGreatherThan(version)) {
+          if (useLegacyOutputFormat) {
```

Comment:
I'm not sure.  I'm guessing nothing is doing it automatically but I wasn't sure if maybe people are parsing the report with another tool.  Since it was only a few lines of code to add the old format I figured it was worth keeping for one more release.

Reply:
Fair enough.